Content.Client/Storage/Visualizers/MappedItemVisualizer.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections.Generic;
+using Content.Shared.Storage;
+using JetBrains.Annotations;
+using Robust.Client.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.Utility;
+
+namespace Content.Client.Storage.Visualizers
+{
+    [UsedImplicitly]
+    public class MappedItemVisualizer : AppearanceVisualizer
+    {
+        [DataField("spriteLayers")] private List<string> _spriteLayers = new();
+        [DataField("sprite")] private ResourcePath? _spritePath;
+
+        public override void InitializeEntity(IEntity entity)
+        {
+            base.InitializeEntity(entity);
+
+            if (entity.TryGetComponent<ISpriteComponent>(out var spriteComponent))
+            {
+                _spritePath ??= spriteComponent.BaseRSI!.Path!;
+            }
+        }
+
+        public override void OnChangeData(AppearanceComponent component)
+        {
+            base.OnChangeData(component);
+            InitializeSpriteMap(component);
+            UpdateSprite(component);
```

Comment:
```suggestion
            if(_spriteLayers.Count == 0)
            {
                InitializeSpriteMap(component);
            }
            UpdateSprite(component);
```

then remove both _spriteLayers.Count checks elsewhere